### ISSUE TITLE
Markers don't get unsubscribed after being hidden, if date has previously changed

### ISSUE THREAD

**Issue by gaston-niglia** [namespace-ee/react-calendar-timeline#524]
**Describe the bug**
Let's say you have a _CustomMark_ on your Timeline, also a DateTimePicker that updated the mark date property on selecting a date form the calendar. And let's say you change the mark date more than once and then hide it by clearing the datetime value. This will NOT make the CustomMark to unsubscribe from the _state.markers_ list.

**To Reproduce**
I have locally added some buttons on the main demo to change a state variable which is mapped to the mark's date property. Like this:

![cusrommarker - before](https://user-images.githubusercontent.com/27180527/53519659-81eea080-3ab2-11e9-8c81-87b16b6b920a.gif)

**Expected behavior**
If a Mark gets hidden, I'd expect that Mark to get correctly unsubscribed from the makers list

**Screenshots**
The issue is because once you change the date from the marker, even though is the same marker, is not the exact same object from the prev state, as the date is different so the current code is not picking this up.

This is the marker with a particular date timestamp:

<img width="397" alt="screen shot 2019-02-27 at 4 04 53 pm" src="https://user-images.githubusercontent.com/27180527/53519914-2375f200-3ab3-11e9-99cb-c2565fef8b9a.png">

Then you change the marker's date property through a dateTimePicker for instance:

<img width="400" alt="screen shot 2019-02-27 at 4 05 20 pm" src="https://user-images.githubusercontent.com/27180527/53519923-296bd300-3ab3-11e9-8727-23802fc52dca.png">

Then, if you want to hide the Marker is not gonna get removed from the list due to the following code:

![screen shot 2019-02-27 at 4 09 26 pm](https://user-images.githubusercontent.com/27180527/53520101-95e6d200-3ab3-11e9-984f-ecdaf0625d21.png)

This issue can be fixed by comparing markers by their id's:

`filter(marker => marker.id !== newMarker.id)`

**Library Version**
0.23.0

**Desktop (please complete the following information):**
* OS: macOS
* Browser chrome

**Additional context**
PR [here](https://github.com/namespace-ee/react-calendar-timeline/pull/525) @Ilaiwi

**Comment by Ilaiwi** [comment]
@gaston-niglia Thank you for the great description and the PR. I am not sure what do you mean by hiding it though? you mean removing it completely as in not rendering it or add a css property for it?

**Comment by gaston-niglia** [comment]
Hi @Ilaiwi, thanks for your reply. Sorry for the confusion, by "hiding it" I mean conditionally remove the CustomMarker depending on a condition. e.g:

```
<TimelineMarkers>
     {someConditionHere && (
          <CustomJumpToMarker date={someDateHere} />
     )}
</TimelineMarkers>
```

So when the CustomMarker is not rendering it fires the "componentWillUnmount" which should cause the Marker to unsubscribe from the Calendar.

I hope I'm explaining better. Please let me know if you have any question or concern.

Thanks

**Comment by Ilaiwi** [comment]
@gaston-niglia  thank you. I merged it. I plan to do a release later this week.

**Comment by gaston-niglia** [comment]
Thank you! Much appreciated mate. @Ilaiwi

**Comment by gaston-niglia** [comment]
Hi mate, do you have any ETA for the release? Thanks @Ilaiwi

**Comment by Ilaiwi** [comment]
@gaston-niglia  tomorrow. sorry I've been busy completing the custom headers

**Comment by gaston-niglia** [comment]
No worries at all, I was just asking

**Comment by Ilaiwi** [comment]
released in 0.23.1 thank you for the contribution

**Comment by gaston-niglia** [comment]
Thank you for your time mate!!
